### ISSUE TITLE
[Suggestion / Enhancement] Decreasing memory footprint / tmp dependencies of philosopher database processing

### ISSUE THREAD

**Issue by clb21565** [Nesvilab/philosopher#105]
Dear Philosopher team, 

I just wanted to pass along some issues I've been having / hope some day to resolve. I am developing metaproteomics methods on our institutional HPC, and your lovely workflow has been very useful. However, your database step has a VERY large memory footprint for large databases which has slowed my progress down. For instance, an entire database run for a large db (~100GB) is unable to process on our high memory node and had a memory usage at point of failure of ~1.2 TB:

[clb21565@calogin1 2020_02_19]$ seff 295568
Job ID: 295568
State: OUT_OF_MEMORY (exit code 0)
Nodes: 1
Cores per node: 32
CPU Utilized: 2-05:34:49
CPU Efficiency: 23.59% of 9-11:09:20 core-walltime
Job Wall-clock time: 07:05:55
Memory Utilized: 1.12 TB
Memory Efficiency: 85.25% of 1.31 TB

Also, because many files are written to /tmp/ directories, I have had to bind mount /tmp/ to /scratch-local/ in order to keep the job from failing. 

In short, this is a great package but has some serious pains to work out for metaproteomics. Apart from this, it's probably the best option for metaproteomics available right now. 

Just wanted to pass this along! 

Thanks, 

Connor

**Comment by prvst** [comment]
I'm glad you find our project useful.

Philosopher was mainly designed for traditional shotgun proteomics analysis, meaning that its functions are tuned for that type of data, you have a different scenario here. 

Could you share a little bit more about your project ? things like: type of instrument, type of data, database, how many samples/fractions, etc. (feel free to write directly to me via email if you are not comfortable by sharing these information in public).

I would like to understand better what you are trying to do, maybe we can adapt Philosopher for cases like this.

**Comment by prvst** [comment]
no response
